.github/workflows/release-library-client.yml

```diff
@@ -0,0 +1,54 @@
+name: "release-library-client"
+on:
+  push:
+    tags:
+      - 'client-v[0-9]+.[0-9]+.[0-9]+'
+      - 'client-v[0-9]+.[0-9]+.[0-9]+-[a-z]+'
+      - 'client-v[0-9]+.[0-9]+.[0-9]+-[a-z]+.[0-9]+'
+
+jobs:
+  release:
+    permissions: write-all
+    environment: Maven Central Release
+    runs-on: 'ubuntu-22.04' 
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Free Disk Space
+        uses: jlumbroso/free-disk-space@main
+        with:
+          android: false
+
+      - name: Install nix
+        uses: cachix/install-nix-action@v27
+        with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+          nix_path: nixpkgs=channel:nixos-24.05
+
+      - uses: cachix/cachix-action@v15
+        with:
+          name: holochain-ci
+                
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Generate gradle.properties
+        run: |
+          cp libraries/client/gradle.properties.ci libraries/client/gradle.properties
+
+          base64 -d <<< "${{ secrets.SIGNING_KEY_RING_FILE_B64 }}" >> $RUNNER_TEMP/keyring.gpg
+          echo "signing.secretKeyRingFile=$RUNNER_TEMP/keyring.gpg" >> libraries/client/gradle.properties
+          echo "signing.keyId=${{ secrets.SIGNING_KEY_ID }}" >> libraries/client/gradle.properties
+          echo "signing.password=${{ secrets.SIGNING_KEY_PASSWORD }}" >> libraries/client/gradle.properties
+          echo "mavenCentralUsername=${{ secrets.MAVEN_CENTRAL_USERNAME }}" >> libraries/client/gradle.properties
+          echo "mavenCentralPassword=${{ secrets.MAVEN_CENTRAL_PASSWORD }}" >> libraries/client/gradle.properties
```

Comment:
Is it save to write secrets in plain text to the runner's disk? Would there potentially be a way to pass them as cli arguments instead?

Reply:
Hmm, good question. I think I could pass them as one-off env variables.

I believe a runner is just a docker container, where each run would have have fresh virtual disk, so the security risk could be a subsequent action in the same job reading the file, or a subsequent job on the same physical machine breaking out of its container. Yeah probably the first one is worth avoiding.

Reply:
switched to using env variables: https://github.com/holochain/android-service-runtime/pull/111/commits/c731e2f730e8e56bddbcf1fe9d81b7324d585b70